### ISSUE TITLE
Interop instead of baked-in animations & gestures

### ISSUE THREAD

**Issue by drcmda** [metafizzy/zdog#31]
Referencing this: https://github.com/metafizzy/zdog/issues/28#issuecomment-497478694

It makes the lib larger, creates complexity, puts a fork into adoption. If this would be a small, focussed project it would benefit from existing eco systems, as well as creating one for its own.

3rd-party use-cases could be documented, i've already seen people use GSAP with it for instance. For gestures especially low level access pointers per element (click, over, out, up, down, etc) are more interesting that high-level abstracts (onDrag).

**Comment by boardfish** [comment]
[3rd-party use cases, you say?](https://codepen.io/anon/pen/mYzjqM?editors=0011) Anime.js plays very nicely with it.

**Comment by drcmda** [comment]
Exactly, this is what i meant! 👍So now you have two engines, the internal one is probably not as performant and battle-tested as Anime is, and it takes up bundle space and developer time. 

If it would be openly interop focussed with nice examples that show how to handle animation and so on, i think that would be a benefit. Not only will it encourage people to make an eco system for it (like threejs'es for instance), but it will also avoid feature creep. And the latter will never stop once you start, people will request animations, then gestures, then orbitcontrols, then trackball, airplanecontrols, etc. In the end you have a giant nest of responsibilities that aren't easy to fix and maintain.

**Comment by desandro** [comment]
I'm having trouble understanding what specifically you're requesting: Do you want more docs around how to use Zdog with other libraries? Like for animation?

**Comment by drcmda** [comment]
I would love if it doesn't ship easings, gestures, etc. But focus on pure interop, so that an eco system can evolve around it. Right now it puts high level over low level interfaces. There's a high level gesture dragger available for instance, but no pointer access events. I would throw out the gesture all together, add basic events, and put the dragger into a separate control. This would also encourage people to write more controls and extensions.

**Comment by desandro** [comment]
Okay. Yeah, I think we're on the same page. My aim was to make Zdog fairly lightweight. But in the process of its creation, making dozens of demos, I found there were some common patterns I kept re-using: like Illustration, Dragger, easeInOut. These non-essential patterns made it in. Experience developers may not need them, but for novice or lazy developers (like myself), they are convenient.

But you're not required to. I've already seen Zdog used with anime.js, [GreenSock](https://codepen.io/chrisgannon/pen/wbYWdj), [D3](https://codepen.io/SgiobairOg/pen/QRobQW). So I think Zdog is a good middle ground. Thanks for this feedback!
